doc/adr/0007-queryable-properties.md

```diff
@@ -0,0 +1,96 @@
+# 7. Queryable properties
+
+Date: 2022-10-13
+
+## Status
+
+New
+
+## Context
+
+User of the trace-viewer application are often interested in additional information from the data provider. For example, the Trace Compass latency statistics views has columns about minimum and maximum duration and user would like to navigate to the time range where this maximum occurred. The front-end that uses the Trace Server Protocol doesn't know about what data it displaying. It only knows about different graph types. For the latency views it's a table tree with columns, but it doesn't know that there is a minimum column or maximum column, nor it knows where in the trace it occurred.
+
+The Trace Server Protocol (TSP) needs to be augmented to provide such type of data, so that the front-end can provide controls for the user to navigate.
+
+**Use cases:**
+
+- GoTo min/max per entry or all
+- Get longest executions per entry or all
+- Get the N longest execution per thread
+- Get Crash details (time and other details like CPU)
+- GoTo most utilized frame
+- Get the N most utilised spans in order
+- Run critical path analysis (returns a data provider descriptor)
+
+The challange is to handshake these properties or capabilites between front-end (clients) and server over an domain-agnostic protocol.
+
+### Data Structure
+
+Augment the `DataProviderDescriptor` data structure with information about queryable properties.
+
+```java
+    /*
+     * QuerablePropertyDescriptor {
+     *  name:        string
+     *  description: string
+     *  id:          string
+     *  actionType:  apply, display
```

Comment:
Can we add some more details on the meaning of "actionType"? By reading this ADR it is not very clear to me when the server is supposed to return one type or the other. Maybe also adding some info to line 73 below could help to clarify.

Same for the input type.